### PR TITLE
Machine objects for master nodes contain libvirt providerSpec

### PR DESCRIPTION
**Pull request by sadasu**
Fixes issue #32

### CODE REVIEW

File: pkg/asset/machines/baremetal/machines.go [hunk 1]
```diff
@@ -61,6 +60,5 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 }
 
 func provider(clusterName string, networkInterfaceAddress string, platform *baremetal.Platform, userDataSecret string) *libvirtprovider.LibvirtMachineProviderConfig {
-	// FIXME: baremetal
-	return &libvirtprovider.LibvirtMachineProviderConfig{}
+	return &baremetalprovider.BareMetalMachineProviderSpec{}
```
Comment (stbenjam) [inline pkg/asset/machines/baremetal/machines.go:63]: My guess is we should populate the fields in BareMetalMachineProviderSpec, like other providers do: https://github.com/openshift-metal3/kni-installer/blob/master/pkg/asset/machines/aws/machines.go#L78-L111, but I'm not 100% sure.

**Comment by stbenjam** [comment]
Could you also run `go fmt` on your code base? Looks like some of the indentation is off

File: pkg/asset/machines/baremetal/machines.go (hunk 1 above)
Comment (russellb) [reply pkg/asset/machines/baremetal/machines.go:63]: Yeah, though we don't necessarily have to block on that for the first PR.  The current cluster-api-provider-baremetal we're running will ignore it either way.

We need to populate some fields here before we can rebase to 4.2 though, as that will include the version that expects some data in here

**Comment by sadasu** [comment]
Removing WIP because my local testing went fine.

**Comment by russellb** [comment]
Great!  I'll give this a quick local test, as well

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/664/

**Comment by russellb** [comment]
This looks good in my local testing, and the CI failure doesn't seem related

**Comment by russellb** [comment]
@sadasu The only thing remaining is to update the commit message.  You updated the PR title, but the commit messages still says WIP.

You can do this locally with `git commit --amend`.  Then push the update to your branch with `git push -f origin sdasu-dev`

**Comment by sadasu** [comment]
@russellb Done with fixing the commit message and kicked off another CI run.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/668/

**Pull request merged**